### ISSUE TITLE
[Dynamo] Unexpected fallback behavior for `aten._to_copy`

### ISSUE THREAD

**Issue by seanlatias** [pytorch/xla#6288]
## 🐛 Bug

I tried to run `torch.compile(backend='openxla')` with a simple gpt2 model from transformers. The program errors out with `Input tensor is not an XLA tensor: CPUFloatType`. After some investigation, it is because `aten._to_copy.default` is categorized as a fallback op because it has non-xla tensors as inputs. This makes the graph ends up with a partition that is not accepted by the compiler, which requires the outputs should be on an XLA device. Following is an example of partitioned graph

```
def forward(...):
    fused_0 = self.fused_0()
    getitem_111 = fused_0[0]
    getitem_112 = fused_0[1]
    getitem_113 = fused_0[2]
...
    _to_copy_4 = torch.ops.aten._to_copy.default(getitem_111, dtype = torch.float32, layout = torch.strided, device = device(type='xla', index=0));  getitem_111 = None
    _to_copy_12 = torch.ops.aten._to_copy.default(getitem_112, dtype = torch.float32, layout = torch.strided, device = device(type='xla', index=0));  getitem_112 = None
    _to_copy_7 = torch.ops.aten._to_copy.default(getitem_113, dtype = torch.float32, layout = torch.strided, device = device(type='xla', index=0));  getitem_113 = None
...
```

And this is how `fused_0` looks like.

```
def forward(self):
    full_7 = torch.ops.aten.full.default([], -3.4028234663852886e+38, dtype = torch.float32, device = device(type='cpu'), pin_memory = False)
    full_23 = torch.ops.aten.full.default([], -3.4028234663852886e+38, dtype = torch.float32, device = device(type='cpu'), pin_memory = False)
    full_13 = torch.ops.aten.full.default([], -3.4028234663852886e+38, dtype = torch.float32, device = device(type='cpu'), pin_memory = False)
...
    return (full_7, full_23, full_13, ...)
```

As can be seen, in the fused graph, the output tensors are not on xla device, which leads to the error.

## To Reproduce

```python
import torch_xla.core.xla_model as xm
import torch
from transformers import AutoTokenizer, GPT2Model

name = 'gpt2'
tokenizer = AutoTokenizer.from_pretrained(name, padding_side='left')
tokenizer.pad_token_id = tokenizer.eos_token_id

model = GPT2Model.from_pretrained("gpt2")
print(model)

model = model.to(xm.xla_device())
model.eval()

batch_size = 1
sequence_length = 128

with torch.no_grad():
    einputs = torch.randint(255, (batch_size, sequence_length), dtype=torch.int).to(xm.xla_device())
    eattn = torch.zeros(batch_size, sequence_length, dtype=torch.int).to(xm.xla_device())

    model = torch.compile(model, backend='openxla')
    out = model(input_ids=einputs, attention_mask=eattn)
```

## Expected behavior

Currently, we filtered out the ops having non-xla-tensor inputs as unsupported ops. The reason why I included this logic is if the fused graph after partition contains such ops, then the non-xla tensors can become inputs to the fused graph, which violates this [checking](https://github.com/pytorch/xla/blob/master/torch_xla/core/dynamo_bridge.py#L215-L222). However, now the issue is that the fused graph have non-xla outputs, which also ends up with an error. There are several ways we con solve this issue.

1. Need to know why there are combinations of `full` + `_to_copy`. If the `full` op has the right device starting from the beginning, we won't end up here. Maybe this is related to how `.to()` work?
2. We unfuse the partitioned graph if the fused graph ends up having non-xla tensors as inputs or outputs. However, this could hurt the performance because we may end up having a graph with no partition.
3. We white-list the combination of `full` + `_to_copy`. The downside of this approach is that it is too hacky.
4. Any other suggestions?

## Environment

 - Reproducible on XLA backend: CPU
 - torch_xla version: 2.1.0

## Additional context

CC: @wonjoolee95 @JackCaoG

**Comment by JackCaoG** [comment]
@ysiraichi can you take a look? The goal is to emit a graph that's as complete as possible

**Comment by ysiraichi** [comment]
#5922 should've fixed that. @seanlatias could you try it with `master` ?

**Comment by seanlatias** [comment]
Hmmm, I tried to cherry-pick but it seems `ConstructorMoverPass` is also not available in PyTorch 2.1.2 also. It's only available with the latest PyTorch. However, it seems your solution can solve the issue and I can have some local workaround. I'll comeback and apply your solution once PT2.2 is alive. Thanks.
